In [1]:
# Library
import sys
import os

# Add the parent directory to sys.path so 'src' can be imported
sys.path.append(os.path.abspath(os.path.join(os.getcwd(), '..')))

In [2]:
# Import Functions
from src.preprocessing import (
    load_data,
    show_shape,
    preview,
    info,
    summary,
    check_missing,
    check_duplicates,
)

Data processing module loaded successfully.


### Load Dataset

In [4]:
df = load_data('D:/Data Rafdhi/Programming/Programming Project (Git)/DBSCAN-Country-Clustering/data/Country-data.csv')

Data loaded successfully from D:/Data Rafdhi/Programming/Programming Project (Git)/DBSCAN-Country-Clustering/data/Country-data.csv


### Shape of Dataset

In [5]:
show_shape(df)

(167, 10)

The dataset contains information about 167 countries with 10 variables.

### Preview Dataset

In [6]:
preview(df)

country  child_mort  exports  health  imports  income  \
0          Afghanistan        90.2     10.0    7.58     44.9    1610   
1              Albania        16.6     28.0    6.55     48.6    9930   
2              Algeria        27.3     38.4    4.17     31.4   12900   
3               Angola       119.0     62.3    2.85     42.9    5900   
4  Antigua and Barbuda        10.3     45.5    6.03     58.9   19100   

   inflation  life_expec  total_fer   gdpp  
0       9.44        56.2       5.82    553  
1       4.49        76.3       1.65   4090  
2      16.10        76.5       2.89   4460  
3      22.40        60.1       6.16   3530  
4       1.44        76.8       2.13  12200

### Summary of Dataset

In [7]:
info(df)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 167 entries, 0 to 166
Data columns (total 10 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   country     167 non-null    object 
 1   child_mort  167 non-null    float64
 2   exports     167 non-null    float64
 3   health      167 non-null    float64
 4   imports     167 non-null    float64
 5   income      167 non-null    int64  
 6   inflation   167 non-null    float64
 7   life_expec  167 non-null    float64
 8   total_fer   167 non-null    float64
 9   gdpp        167 non-null    int64  
dtypes: float64(7), int64(2), object(1)
memory usage: 13.2+ KB


The dataset including economic and health indicators such as child mortality, exports, health expenditure, imports, income, inflation, life expectancy, total fertility rate, and GDP per capita. Each variable provides insights into the socio-economic status and development level of each country. The data is complete with no missing values, allowing for comprehensive analysis and comparison across countries.

### Summary Statistics of Dataset

In [8]:
summary(df)

child_mort     exports      health     imports         income  \
count  167.000000  167.000000  167.000000  167.000000     167.000000   
mean    38.270060   41.108976    6.815689   46.890215   17144.688623   
std     40.328931   27.412010    2.746837   24.209589   19278.067698   
min      2.600000    0.109000    1.810000    0.065900     609.000000   
25%      8.250000   23.800000    4.920000   30.200000    3355.000000   
50%     19.300000   35.000000    6.320000   43.300000    9960.000000   
75%     62.100000   51.350000    8.600000   58.750000   22800.000000   
max    208.000000  200.000000   17.900000  174.000000  125000.000000   

        inflation  life_expec   total_fer           gdpp  
count  167.000000  167.000000  167.000000     167.000000  
mean     7.781832   70.555689    2.947964   12964.155689  
std     10.570704    8.893172    1.513848   18328.704809  
min     -4.210000   32.100000    1.150000     231.000000  
25%      1.810000   65.300000    1.795000    1330.000000  
50%      5.390000   73.100000    2.410000    4660.000000  
75%     10.750000   76.800000    3.880000   14050.000000  
max    104.000000   82.800000    7.490000  105000.000000

### Check for Missing Value

In [9]:
check_missing(df)

country       0
child_mort    0
exports       0
health        0
imports       0
income        0
inflation     0
life_expec    0
total_fer     0
gdpp          0
dtype: int64

### Check for Duplicated Data

In [10]:
check_duplicates(df)

Number of duplicated rows: 0
